frontend/src/app/features/work-packages/routing/wp-view-base/state/wp-single-view.service.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  Injectable,
+  Injector,
+} from '@angular/core';
+import { WpSingleViewStore } from './wp-single-view.store';
+import { WpSingleViewQuery } from 'core-app/features/work-packages/routing/wp-view-base/state/wp-single-view.query';
+import {
+  filter,
+  map,
+  switchMap,
+  take,
+} from 'rxjs/operators';
+import {
+  selectCollection$,
+  selectCollectionEntities$,
+} from 'core-app/core/state/collection-store.type';
+import { InAppNotification } from 'core-app/core/state/in-app-notifications/in-app-notification.model';
+import { InAppNotificationsService } from 'core-app/core/state/in-app-notifications/in-app-notifications.service';
+import { ApiV3ListFilter } from 'core-app/core/apiv3/paths/apiv3-list-resource.interface';
+import {
+  markNotificationsAsRead,
+  notificationsMarkedRead,
+} from 'core-app/core/state/in-app-notifications/in-app-notifications.actions';
+import { ActionsService } from 'core-app/core/state/actions/actions.service';
+import { UntilDestroyedMixin } from 'core-app/shared/helpers/angular/until-destroyed.mixin';
+import {
+  EffectCallback,
+  EffectHandler,
+} from 'core-app/core/state/effects/effect-handler.decorator';
+
+@EffectHandler
+@Injectable()
+export class WpSingleViewService extends UntilDestroyedMixin {
+  id = 'WorkPackage Activity Store';
+
+  protected store = new WpSingleViewStore();
+
+  readonly query = new WpSingleViewQuery(this.store);
+
+  selectNotifications$ = this
+    .query
+    .select((state) => state.notifications.filters)
```

Comment:
I think I'd prefer a `selectFilters$` on this query instead of doing it here

Reply:
I think all selection logic that can be in the query should be there. This service would get really bloated if at some point we have the same kind of stores for work packages, and you're selecting from multiple different collections and stores. The less code there is here, the better. It also makes it easier to reuse query logic.

Reply:
Yes, that makes sense. The only downside of this is that you need to pass in the ianService to the query (to get `selectCollectionAsEntities$` for example).

I changed that accordingly